### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by fuadmoin**
In this branch I:

- [ ] used ruby gems as a software package system.
- [ ] install Ruby on Rails framework.
- [ ] used controllers to handle requests and render views.
- [ ] used params from browser requests in a safe way.
- [ ] used preprocessed HTML file with embedded Ruby code.
- [ ] used the navbar layouts for shared content.
- [ ] used database migration files to maintain database schema.
- [ ] used validation for the models.
- [ ] secured app from n + 1 problems.
- [ ] set up associations between the models.
- [ ] built a web app that requires the user to log in.
- [ ] used tailwind CSS to style the app.
- [ ] user can create categories.
- [ ] user can create items and associate them with categories.
- [ ] created splash screen.
- [ ] created a navbar.
- [ ] recorded a video and attached it to the readme.
- [ ] deployed the project.
- [ ] updated the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,26 +31,50 @@
...
-- **uses react**
+- **Keeps track of all your items and categories**
+- **Allows a user to register**
+- **Allows a user to add new items and categories**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
+### Video Presentation <a name="presentation"></a>
+
+<a href="https://www.loom.com/share/efb4c27a5a0040af81f26c37b383531d">Watch video</a>
```
Comment (TadesseDev) [inline README.md:73]: - [Optional] Dear @fuadmoin, Your video is clear and precise. What bothers me a bit was, the volume level, It was too low I had to set my volume to max and till I couldn't hear it well, Please consider re-recording the video documentation.

File: app/views/categories/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<div class="items-container rounded-md bg-gray-100  p-3  shadow-sm ">
+<%= form_for @category do |f| %>
+<%= f.text_field :name, placeholder: "Enter Category's name", class:"block all-in-one-label   round-md bg-transparent py-1.5 pl-1  sm:text-sm sm:leading-6 " %>
```
Comment (TadesseDev) [inline app/views/categories/new.html.erb]: - [ ] Dear @fuadmoin, while creating a category, Please make sure the app can stand to end user error, the user might submit the form without filling in the required/mandatory fields, and this is causing an issue in your case. Your app crashes in such a case. Please apply HTML5 form validation to make these fields required and overcome the issues. 

### Submitting form without filling it 
image

- [ ] These same issues happen on the `Add a new transaction/item`, Please take a look at this too.  And here, its mandatory to check at least one `category`, beside the others.
> It's recommended we always validate all user input to make sure that anyone with bad intentions cannot compromise our app, both in front and backend.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @fuadmoin 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Good job overall following the template design ✅
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @fuadmoin, 






Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice design 🔥 
✅ Nice implementing the changes 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**